std/functional.d

```diff
@@ -260,6 +210,83 @@ unittest
     static assert(!is(typeof(binaryFun!FuncObj)));
 }
 
+/**
+Transforms a string representing an expression into a n-ary function.
+The number of arguments is given as $(argc).
+The string must either use symbol names $(D a-z) as the parameters or
+provide the symbols via the $(D parmNames) arguments.
+If $(D fun) is not a string, $(D naryFun) aliases itself away to $(D fun).
+*/
+
+template naryFun(size_t argc, alias fun, parmNames...)
```

Comment:
Unless `paramNames` isn't given. Then we need `argc`. I guess I could use an overload instead.

On December 20, 2015 8:50:35 PM EST, JakobOvrum <notifications@github.com> wrote:
>> @@ -260,6 +210,83 @@ unittest
>>      static assert(!is(typeof(binaryFun!FuncObj)));
>>  }
>>  
>> +/**
>> +Transforms a string representing an expression into a n-ary
>function.
>> +The number of arguments is given as $(argc).
>> +The string must either use symbol names $(D a-z) as the parameters
>or
>> +provide the symbols via the $(D parmNames) arguments.
>> +If $(D fun) is not a string, $(D naryFun) aliases itself away to $(D
>fun).
>> +*/
>> +
>> +template naryFun(size_t argc, alias fun, parmNames...)
>
>Is `argc` really necessary? Looks like `parmNames.length`.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/D-Programming-Language/phobos/pull/3882/files#r48112338